src/main/scala/scorex/core/network/peer/PeerDatabaseImpl.scala

```diff
@@ -10,13 +10,13 @@ import scala.collection.JavaConverters._
 
 
 //todo: persistence
-class PeerDatabaseImpl(settings: NetworkSettings,
-                       filename: Option[String],
-                       timeProvider: NetworkTimeProvider) extends PeerDatabase {
+class PeerDatabaseImpl(bindAddress: InetSocketAddress,
+                       declaredAddress: Option[InetSocketAddress],
+                       filename: Option[String]) extends PeerDatabase {
```

Comment:
Do we really want to use a global `TimeProvider` instead of passing locally as an argument?